test/performance/performance-tests.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script runs the end-to-end tests against Knative Serving built from source.
+# It is started by prow for each PR. For convenience, it can also be executed manually.
+
+# If you already have a Kubernetes cluster setup and kubectl pointing
+# to it, call this script with the --run-tests arguments and it will use
+# the cluster and run the tests.
+
+# Calling this script without arguments will create a new cluster in
+# project $PROJECT_ID, start knative in it, run the tests and delete the
+# cluster.
+
+source $(dirname $0)/../e2e-common.sh
+
+# Skip installing istio as an add-on.
+# Temporarily increasing the cluster size for serving tests to rule out
+# resource/eviction as causes of flakiness.
+initialize --skip-istio-addon --min-nodes=4 --max-nodes=4 --enable-ha --perf --cluster-version=1.21 "$@"
+
+header "Updating cluster"
+
+# Update the activator hpa minReplicas to 10
+kubectl patch hpa -n "${SYSTEM_NAMESPACE}" activator --patch '{"spec": {"minReplicas": 10}}'
+
+# Update the scale-to-zero grace period to 10s
+kubectl patch configmap/config-autoscaler -n "${SYSTEM_NAMESPACE}" \
+    --type merge \
+    -p '{"data":{"scale-to-zero-grace-period":"10s"}}'
+
+# Ensure gradual rollout is enabled.
+kubectl patch configmap/config-network -n "${SYSTEM_NAMESPACE}"\
+    --type merge \
+    -p '{"data":{"rolloutDuration":"240"}}'
```

Comment:
We can make these kubectl patch's overlays in the performance directory. Just make sure the perf overlays are applied after the HA ones

Reply:
Following up

1) can we lookup what the previous configuration was for our performance cluster and match that? 
2) `-p '{"data":{"rolloutDuration":"240"}}'` curious why we're setting this 
3) We should also disable the chaos duck - as we don't want controllers etc to disappear when running these tests

Reply:
This is what I did, the `-p '{"data":{"rolloutDuration":"240"}}'` line is from our previous perf cluster setup

Reply:
The perf cluster when using mako was created using this code here: https://github.com/knative/test-infra/blob/5273590479dfaa642f6a37e6dbb0bcb182047847/pkg/clustermanager/perf-tests/main.go and this config: https://github.com/knative/serving/blob/77ba3bdc0f0e3a045783dfe7b61283dd4d80eb31/test/performance/benchmarks/scale-from-zero/cluster.yaml

then the following applied to it:
https://github.com/knative/serving/blob/77ba3bdc0f0e3a045783dfe7b61283dd4d80eb31/test/performance/performance-tests.sh#L40-L82

Reply:
> from our previous perf cluster setup

Ok cool - let's still move the kubectl patch commands to ytt overlay files

Reply:
moved to overlays